matgen/generate_matrix_tz.cc

```diff
@@ -0,0 +1,331 @@
+                d_one, d_one, Sigma.data(), Sigma.size() );
+            break;
+
+        case TestMatrixType::jordan: {
+            set(zero, one, A ); // ones on diagonal
+            if (A.uplo() == Uplo::Lower) {
+                // ones on sub-diagonal
+                for (int64_t i = 0; i < nt; ++i) {
+                    // Set 1 element from sub-diagonal tile to 1.
+                    if (i > 0) {
+                        if (A.tileIsLocal(i, i-1)) {
+                            A.tileGetForWriting( i, i-1, LayoutConvert::ColMajor );
+                            auto T = A(i, i-1);
+                            T.at(0, T.nb()-1) = 1.;
+                        }
+                    }
+
+                    // Set 1 element from sub-diagonal tile to 1.
+                    if (A.tileIsLocal(i, i)) {
+                        A.tileGetForWriting( i, i, LayoutConvert::ColMajor );
+                        auto T = A(i, i);
+                        auto len = T.nb();
+                        for (int j = 0; j < len-1; ++j) {
+                            T.at(j+1, j) = 1.;
+                        }
+                    }
+                }
+            }
+            else { // upper
+                // ones on sub-diagonal
+                for (int64_t i = 0; i < nt; ++i) {
+                    // Set 1 element from sub-diagonal tile to 1.
+                    if (i > 0) {
+                        if (A.tileIsLocal(i-1, i)) {
+                            A.tileGetForWriting( i-1, i, LayoutConvert::ColMajor );
+                            auto T = A(i-1, i);
+                            T.at(T.nb()-1, 0) = 1.;
+                        }
+                    }
+
+                    // Set 1 element from sub-diagonal tile to 1.
+                    if (A.tileIsLocal(i, i)) {
+                        A.tileGetForWriting( i, i, LayoutConvert::ColMajor );
+                        auto T = A(i, i);
+                        auto len = T.nb();
+                        for (int j = 0; j < len-1; ++j) {
+                            T.at(j, j+1) = 1.;
+                        }
+                    }
+                }
+            }
+            break;
+        }
+
+        case TestMatrixType::rand:
+        case TestMatrixType::rands:
+        case TestMatrixType::randn:
+        case TestMatrixType::randb:
+        case TestMatrixType::randr: {
+            auto rand_dist = slate::random::Dist(int(type));
+            #pragma omp parallel
+            #pragma omp master
+            {
+                int64_t j_global = 0;
+                for (int64_t j = 0; j < nt; ++j) {
+                    int64_t i_global = 0;
+                    // lower trapezoid is [ j .. mt )
+                    // upper trapezoid is [ 0 .. min( j+1, mt ) )
+                    int64_t i_start = A.uplo() == Uplo::Lower ? j  : 0;
+                    int64_t i_end   = A.uplo() == Uplo::Lower ? mt : std::min( j+1, mt );
```

Comment:
Could you add the following snippet right after this line?  I've been meaning to make a PR for this, but we may as well just rolled it into this PR.
```
for (int64_t i = 0; i < i_start; ++i) {
    i_global += A.tileMb( i );
}
```

Reply:
Perfect, thanks.